ports/javascript/wrapper.js

```diff
@@ -71,7 +71,42 @@ var mainProgram = function()
       } else {
           mp_js_do_str(contents);
       }
+   } else {
+     initBrowser();
+   }
+}
+
+Module["onRuntimeInitialized"] = mainProgram;
+
+var __initBrowserCount = 2;
+
+function initBrowser(e)
+{
+  __initBrowserCount--;
+  // window.onLoad and wasm have loaded...
+  if (__initBrowserCount == 0)
+  {
+    var mp_js_stdout = document.getElementById('mp_js_stdout');
+    mp_js_stdout.addEventListener('print', function(e) {
```

Comment:
I think the reason this was done in the html before is so it can be customised for the particular use/app.  I guess this default behaviour can still be overridden by a particular app, by changing the event listener?